src/amberscript/parser.cc

```diff
@@ -2625,14 +2634,18 @@ Result Parser::ParseDebug() {
     }
   }
 
-  command_list_.back()->SetDebugScript(std::move(dbg));
+  cmd->SetDebugScript(std::move(dbg));
 
   return Result();
 }
 
-Result Parser::ParseDebugThread(debug::Events* dbg) {
+Result Parser::ParseDebugThread(debug::Events* dbg, Pipeline* pipeline) {
   auto token = tokenizer_->NextToken();
   if (token->AsString() == "GLOBAL_INVOCATION_ID") {
+    for (auto& shader : pipeline->GetShaders()) {
+      shader.SetEmitDebugInfo(true);
```

Comment:
The other two places this is done guard on shader type, should this do the same?

Reply:
`GLOBAL_INVOCATION_ID` isn't tied to a specific shader type, so no.